overlord/fdestate/fdestate.go

```diff
@@ -204,26 +206,32 @@ func initializeState(st *state.State) error {
 	devpData := fmt.Sprintf("/dev/disk/by-uuid/%s", dataUUID)
 	devpSave := fmt.Sprintf("/dev/disk/by-uuid/%s", saveUUID)
 	digest, err := getPrimaryKeyDigest(devpData)
-	if err != nil {
-		return fmt.Errorf("cannot obtain primary key digest for data device %s: %w", devpData, err)
-	}
-	// TODO: restore key verification once we know that it is always added to
-	// the keyring
-	sameDigest, err := digest.verifyPrimaryKeyDigest(devpSave)
 	if err != nil {
 		if !errors.Is(err, secboot.ErrKernelKeyNotFound) {
-			return fmt.Errorf("cannot verify primary key digest for save device %s: %w", devpSave, err)
-		} else {
+			return fmt.Errorf("cannot obtain primary key digest for data device %s: %w", devpData, err)
+		}
+		// Some very old kernels with FDE do not set primary key in the keyring
+		logger.Noticef("cannot obtain primary key digest for data device %s: %v", devpData, err)
+		s.PrimaryKeys = map[int]PrimaryKeyInfo{}
```

Comment:
won't this blow up later when KeyslotRoles entries mention primary key ID 0 but it's not there?

Reply:
I am just this will be a lot less dramatic. And they already do. But we do not really use that information. It is a digest. Just to identify the primary key. We cannot read it from there.